pkg/server/container_start.go

```diff
@@ -142,24 +134,36 @@ func (c *criService) startContainer(ctx context.Context,
 	return nil
 }
 
-// Create container loggers and return write closer for stdout and stderr.
-func createContainerLoggers(logPath string, tty bool) (stdout io.WriteCloser, stderr io.WriteCloser, err error) {
+// createContainerLoggers creates container loggers and return write closer for stdout and stderr.
+func (c *criService) createContainerLoggers(logPath string, tty bool) (stdout io.WriteCloser, stderr io.WriteCloser, err error) {
 	if logPath != "" {
 		// Only generate container log when log path is specified.
-		if stdout, err = cio.NewCRILogger(logPath, cio.Stdout); err != nil {
-			return nil, nil, errors.Wrap(err, "failed to start container stdout logger")
+		f, err := os.OpenFile(logPath, os.O_CREATE|os.O_APPEND|os.O_WRONLY, 0640)
+		if err != nil {
+			return nil, nil, errors.Wrap(err, "failed to open log file")
 		}
 		defer func() {
 			if err != nil {
-				stdout.Close()
+				f.Close()
```

Comment:
in the error case I don't think we need to close "f" ...

Reply:
Why not?

Reply:
That error is checked before this defer.

Reply:
exactly and the only way the function can return error (atm) is if it failed to open.. see "return" statements :-)

Reply:
ok.. yeah that's true that it could become useful in the future..